jumpscale/core/base/fields.py

```diff
@@ -105,5 +106,163 @@ def validate(self, value):
             self.field.validate(item)
 
 
+class Email(Field):
+    def __init__(self, default="", **kwargs):
+        super().__init__(default, **kwargs)
+        self.regex = r"^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$"
+
+    def validate(self, value):
+        """Check whether provided value is a valid email representation
+        Args:
+            value (str)
+        Returns:
+            Boolean expresion"""
+        super().validate(value)
+        return re.search(self.regex, value) is not None
+
+
+class Path(Field):
+    def __init__(self, default="", **kwargs):
+        super().__init__(default, **kwargs)
+        self.regex = r"^(/[^/ ]*)+/?$"
+
+    def validate(self, value):
+        """Check whether provided value is a valid path representation
+        Args:
+            value (str)
+        Returns:
+            Boolean expresion"""
+        super().validate(value)
+        return re.search(self.regex, value) is not None
+
+
+class URL(Field):
+    def __init__(self, default="", **kwargs):
+        super().__init__(default, **kwargs)
+        self.regex = r"^(https?|ftp)://[^\s/$.?#].[^\s]*$"
+
+    def validate(self, value):
+        """Check whether provided value is a valid URL representation
+        Args:
+            value (str)
+        Returbs:
+            Boolean expresion"""
+        super().validate(value)
+        return re.search(self.regex, value) is not None
+
+
+class Tel(Field):
+    def __init__(self, default="", **kwargs):
+        super().__init__(default, **kwargs)
+        self.regex = r"^\+?[0-9]{6,15}(?:x[0-9]+)?$"
+
+    def validate(self, value):
+        """Check whether provided value is a valid Telephone number representation
+        Args:
+            value (str)
+        Returbs:
+            Boolean expresion"""
+        super().validate(value)
+        value = self._clean(value)
+        return re.search(self.regex, value) is not None
+
+    def _clean(self, value):
+        if value is not None:
+            value = value.replace(",", "")
+            value = value.replace("-", "")
+            value = value.replace("(", "")
+            value = value.replace(")", "")
+            value = value.replace(" ", "")
+            return value
+
+
+class IPAddress(Field):
+    def __init__(self, default="", **kwargs):
+        super().__init__(default, **kwargs)
+
+    def validate(self, value):
+        """Check whether provided value is a valid IPaddress representation
+        Args:
+            value (str)
+        Returns:
+            Boolean expresion"""
+        import ipaddress
+
+        super().validate(value)
+        try:
+            ipaddress.IPv4Address(value)
```

Comment:
works only for ip4? what about ip6? is that also the same type to use for network?

Reply:
done using ```ipaddress.ip_interface```